scripts/lib/CIME/case/case_setup.py

```diff
@@ -152,10 +152,15 @@ def _case_setup_impl(case, caseroot, clean=False, test_mode=False, reset=False):
 
             case.initialize_derived_attributes()
 
-            cost_per_node = 16 if case.get_value("MACH") == "yellowstone" else case.get_value("MAX_MPITASKS_PER_NODE")
+            cost_per_node = case.get_value("MAX_MPITASKS_PER_NODE")
             case.set_value("COST_PES", case.num_nodes * cost_per_node)
-            case.set_value("TOTALPES", case.total_tasks)
-            case.set_value("SMP_PRESENT", case.get_build_threaded())
+            threaded = case.get_build_threaded()
+            case.set_value("SMP_PRESENT", threaded)
+            if threaded and case.total_tasks * case.thread_count > cost_per_node:
+                smt_factor = max(1,int(case.get_value("MAX_TASKS_PER_NODE") / cost_per_node))
+                case.set_value("TOTALPES", case.total_tasks * case.thread_count / smt_factor)
+            else:
+                case.set_value("TOTALPES", case.total_tasks*case.thread_count)
```

Comment:
If MAX_MPITASKS_PER_NODE=64, MAX_TASKS_PER_NODE=256, 128 total tasks on 2 64-core KNL nodes, NTHRDS=2, then TOTALPES=128*2/4=64 ?

Would it be better to count TOTALPES based on number of nodes multipled by MAX_MPITASKS_PER_NODE? Core-hours are usually charged based on whole number of nodes (a partially-occupied node is charged as a whole node). And there is typically a 1-to-1 correspondence between number of physical cores and MAX_MPITASKS_PER_NODE.

Reply:
COST_PES is total number of nodes.   I am trying to clearly distinguish TOTALPES as the total number of cores used.

Reply:
Be explicit:  TOTAL_CORES_AVAIL, TOTAL_CORES_USED.   Don't name it based on how its used (in calculating a cost).